### PR TITLE
[Unity][Op] Extend `relax.op.take` op to match behavior with `topi.take`. 

### PR DESCRIPTION
**Pull request by sunggg**
Currently, `relax.op.take` implements semantics of [torch.take](https://pytorch.org/docs/stable/generated/torch.take.html) which has a stronger restriction than its backing `topi.take` which seems to follow the semantics of [numpy.take](https://numpy.org/doc/stable/reference/generated/numpy.take.html). 
i.e., `relax.op.take` enforces its input `indices` to be 1-d tensor. 

To match its behavior with `topi.take`, this PR extends the semantics of `relax.op.take` by removing such restriction without breaking existing test cases (except one for checking 1-d tensor restriction). 

As a result, this PR can support [onnx.gather](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Gather-13) which was the main motivation of my previous PR https://github.com/apache/tvm/pull/14457 by resolving the problem in my previous stab. 

cc. @jwfromm @yongwww @MasterJH5574

### CODE REVIEW

File: src/relax/op/tensor/index.cc [hunk 1]
```diff
@@ -47,11 +47,10 @@ StructInfo InferStructInfoTake(const Call& call, const BlockBuilder& ctx) {
...
-  if (indices_sinfo->ndim != 1) {
-    ctx->ReportFatal(Diagnostic::Error(call)
-                     << "Take op requires the input indices to be 1-dimensional tensor. However, "
-                        "the given indices ndim is "
-                     << indices_sinfo->ndim);
+
+  if (indices_sinfo->IsUnknownDtype()) {
+    // TODO(tvm-team): Do we have an equivalent of `ctx->ReportFatal` for warning?
+    LOG(WARNING) << "Data type of indice has not been specified. Assume it has an integer type.";
   } else if (!indices_sinfo->IsUnknownDtype() &&
```
Comment (yongwww) [inline src/relax/op/tensor/index.cc:54]: remove dup check `!indices_sinfo->IsUnknownDtype() &&`

**Review by yongwww — COMMENTED** [review]
Overall looks good to me

**Review by psrivas2 — APPROVED** [review]

**Comment by tqchen** [comment]
as long as the existing behavior are preserved, i think we can support multi-dim index

**Comment by sunggg** [comment]
@tqchen I checked that existing tests are passed, so I believe the existing behavior is intact.

**Comment by tqchen** [comment]
@MasterJH5574 would be great if you can help to take a quick look at this one

File: src/relax/op/tensor/index.cc [hunk 2]
```diff
@@ -47,13 +47,11 @@ StructInfo InferStructInfoTake(const Call& call, const BlockBuilder& ctx) {
...
-  if (indices_sinfo->ndim != 1) {
-    ctx->ReportFatal(Diagnostic::Error(call)
-                     << "Take op requires the input indices to be 1-dimensional tensor. However, "
-                        "the given indices ndim is "
-                     << indices_sinfo->ndim);
-  } else if (!indices_sinfo->IsUnknownDtype() &&
-             !(indices_sinfo->dtype.is_int() || indices_sinfo->dtype.is_uint())) {
+
+  if (indices_sinfo->IsUnknownDtype()) {
+    // TODO(tvm-team): Do we have an equivalent of `ctx->ReportFatal` for warning?
```
Comment (MasterJH5574) [inline src/relax/op/tensor/index.cc:52]: Leaving as LOG(WARNING) looks fine to me.

**Review by MasterJH5574 — APPROVED** [review]

**Comment by MasterJH5574** [comment]
@sunggg Sorry for the delay and thanks for the extension!

**Pull request merged**